lib/overcommit/hook_context/pre_commit.rb

```diff
@@ -7,7 +7,21 @@ module Overcommit::HookContext
   # This includes staged files, which lines of those files have been modified,
   # etc. It is also responsible for saving/restoring the state of the repo so
   # hooks only inspect staged changes.
-  class PreCommit < Base
+  class PreCommit < Base # rubocop:disable ClassLength
+    # Returns whether this hook run was triggered by `git commit --amend`
+    def amend?
+      if @amended.nil?
+        cmd = Overcommit::Utils.parent_command
+        @amended = !(/--amend/ =~ cmd).nil?
```

Comment:
While right now there are no other `git` commands that have an `--amend` flag or a flag with `--amend*` as a prefix (that I know of), for the sake of future proofing we should constrain this regex to prevent false positives:

```ruby
/\scommit\s.*--amend(\s|$)/
```

You can also do an early return here since at this point you know this was an amend, saving a shell invocation below. (admittedly a micro-optimization)

```ruby
return @amended if @amended = !(/--amend/ =~ cmd).nil?
```